tests/benchdnn/graph/input_displacer.cpp

```diff
@@ -229,29 +243,92 @@ partition_data_displacer_t::partition_data_displacer_t(
+                    user_set_value = -INFINITY;
+                    cfg_name = "Implicit_causal_mask";
                 } else {
                     // Padding mask.
                     assert(qk_data_offset == 1 || qk_data_offset == 2);
                     // Fill condition depending on qk values tensor to use only
                     // its values, which is equivalent of not using a mask.
+                    filling_type = filling_type_t::fixed_setting;
                     if (qk_data_offset == 1) {
-                        filling_type = filling_type_t::one;
+                        user_set_value = 1.f;
                     } else if (qk_data_offset == 2) {
-                        filling_type = filling_type_t::zero;
+                        user_set_value = 0.f;
                     }
+                    cfg_name = "Explicit_padding_mask";
                 }
             }
 
             if (filling_type == filling_type_t::undef) {
                 BENCHDNN_PRINT(
                         7, "%s\n", "[DISPLACE]: Filling type was not set");
                 break;
+            } else if (filling_type == filling_type_t::fixed_setting) {
+                const std::vector<float> user_set {user_set_value};
+                displace_args_.emplace(causal_mask_lt->id_,
+                        displace_args_t {aop, offset, *causal_mask_lt,
+                                filling_type, {user_set, cfg_name}});
+            } else if (filling_type == filling_type_t::causal_mask) {
+                // Casual mask filling
+                displace_args_.emplace(causal_mask_lt->id_,
+                        displace_args_t {
+                                aop, offset, *causal_mask_lt, filling_type});
             }
+            break;
+        }
+
+        // Fill scalar for bottom-right implicit casual mask
+        if (aop.kind_ == "Add") {
+            const std::string cfg_name = "Bottom_right_implicit_padding_mask";
+            static constexpr int seq_len_q = 0;
+            static constexpr int seq_len_kv = 1;
+
+            auto *aop_out_lt = &aop.out_lts_[0];
+            auto *child_op = &dg_->get_op_by_in_lt(aop_out_lt->id_);
+            if (child_op->kind_ != "Subtract") break;
+
+            auto *child_op_out_lt = &child_op->out_lts_[0];
+            auto *next_child_op = &dg_->get_op_by_in_lt(child_op_out_lt->id_);
+            if (next_child_op->kind_ != "GreaterEqual") break;
 
-            quantize_displace_.emplace(causal_mask_lt->id_,
-                    std::make_tuple(
-                            aop, offset, *causal_mask_lt, filling_type));
+            // The following subtract and greaterEqual must also be a part of
+            // the partition.
+            if (op_ids_set_.find(child_op->id_) == op_ids_set_.end()
+                    || op_ids_set_.find(next_child_op->id_)
+                            == op_ids_set_.end())
+                break;
+
+            const auto set_scalar_displace_args =
```

Comment:
nit: as this pattern doesn't necessarily have to be scalars, please change the variable name here.